File: src/main/java/me/arasple/mc/trmenu/action/acts/ActionSetTitle.java
package me.arasple.mc.trmenu.action.acts;

import me.arasple.mc.trmenu.action.base.AbstractAction;
import me.arasple.mc.trmenu.api.TrMenuAPI;
import me.arasple.mc.trmenu.menu.Menu;
import me.arasple.mc.trmenu.nms.TrMenuNms;
import me.arasple.mc.trmenu.utils.Vars;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2020/2/7 21:05
 */
public class ActionSetTitle extends AbstractAction {

    @Override
    public String getName() {
        return "set(-)?title";
    }

    @Override
    public void onExecute(Player player) {
        Menu menu = TrMenuAPI.getMenu(player);
        if (menu != null) {
            TrMenuNms.setTitle(player, player.getOpenInventory().getTopInventory(), Vars.replace(player, getContent()));
        }
    }

}

File: src/main/kotlin/me/arasple/mc/trmenu/modules/action/impl/menu/ActionSetTitle.kt
package me.arasple.mc.trmenu.modules.action.impl.menu

import me.arasple.mc.trmenu.modules.action.base.Action
import me.arasple.mc.trmenu.utils.Tasks
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2020/3/8 21:53
 */
class ActionSetTitle : Action("set(-)?title") {

    override fun onExecute(player: Player) {
        Tasks.delay(3) {
            val session = getSession(player)
            if (!session.isNull()) {
                val menu = session.menu ?: return@delay
                val layout = session.layout ?: return@delay

                layout.displayInventory(player, getContent(player))
                menu.resetIcons(player, session)
            }
        }
    }

}